functions/db/import_histograms.fn.js

```diff
@@ -0,0 +1,95 @@
+const admin = require('firebase-admin');
+const { BigQuery } = require('@google-cloud/bigquery');
+const functions = require('firebase-functions');
+const { murmurHash128 } = require('murmurhash-native');
+
+try {
+  admin.initializeApp();
+} catch (e) {
+  console.log(e);
+}
+
+async function readAggregates(channel, version) {
+  const bigquery = new BigQuery();
+  const query = `
+    SELECT *
+    FROM \`moz-fx-data-derived-datasets.analysis.client_probe_counts\`
+    WHERE
+      channel = @channel
+      AND app_version = @app_version
+      AND metric = @metric
+      AND os = @os
+  `;
+  const options = {
+    query,
+    location: 'US',
+    params: {
+      channel,
+      app_version: version,
+      metric: 'scalar_content_browser_usage_plugin_instantiated',
+      os: 'Windows_NT'
+    }
+  };
+  const [rows] = await bigquery.query(options);
+
+  // For each row, build a document for inserting into Firestore.
+  const docs = [];
+  let collection = null;
+  let docId = null;
+  let doc = null;
+
+  rows.forEach(row => {
+    collection = `${row.channel}-${row.app_version.split('.')[0]}`;
+    // Hash the dimensions to create a unique reproducible doc ID.
+    // See: https://softwareengineering.stackexchange.com/a/145633 on the
+    // choice of hashing function.
+    docId = murmurHash128(
+      `${row.metric}-${row.app_build_id}-${row.os}-${row.agg_type}`
+    );
+    doc = {
+      metric: row.metric,
+      build_id: row.app_build_id,
+      os: row.os,
+      agg_type: row.agg_type,
+      aggregates: row.aggregates.key_value
+    };
+    docs.push({
+      collection,
+      docId,
+      doc
+    });
+  });
+  return docs;
+}
+
+async function writeDoc(collection, docId, doc) {
+  const db = admin.firestore();
+
+  return db
+    .collection(collection)
+    .doc(docId)
+    .set(doc);
+}
+
+async function importRows() {
+  const docs = await readAggregates('release', '65.0');
+  const promises = docs.map(doc =>
+    writeDoc(doc.collection, doc.docId, doc.doc)
+  );
+  const results = await Promise.all(promises);
+  return results;
+}
+
+// eslint-disable-next-line no-multi-assign
+exports = module.exports = functions.https.onRequest((request, response) => {
```

Comment:
Like before, depending on the platform being targeted, this could be rewritten like:

```javascript
export default functions.https.onRequest((request, response) => {
```

At least I think that works. It may have to be something like this instead:

```javascript
export default () => {
  functions.https.onRequest((request, response) => {
  }
};
```